src/Core/Components/Checkbox/FluentCheckbox.razor.cs

```diff
@@ -0,0 +1,126 @@
+    [Parameter]
+    public ElementReference Element { get; set; }
+
+    /// <summary>
+    /// Gets or sets the checked state of the checkbox
+    /// </summary>
+    [Parameter]
+    public bool? Checked { get; set; }
+
+    /// <summary>
+    /// Gets or sets the indeterminate state of the checkbox
+    /// </summary>
+    [Parameter]
+    public bool? Indeterminate { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the CheckBox will allow three check states rather than two.
+    /// </summary>
+    [Parameter]
+    public bool ThreeState { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating the order of the three states of the CheckBox.
+    /// False(by default), the order is Unchecked -> Checked -> Intermediate.
+    /// True: the order is Unchecked -> Intermediate -> Checked.
+    /// </summary>
+    [Parameter]
+    public bool ThreeStateChecked { get; set; }
+
+    /// <summary>
+    /// Gets or sets the shape of the checkbox
+    /// </summary>
+    [Parameter]
+    public CheckboxShape Shape { get; set; } = CheckboxShape.Square;
+
+    /// <summary>
+    /// Gets or sets the content to prefix the input component.
+    /// </summary>
+    [Parameter]
+    public virtual RenderFragment? StartTemplate { get; set; }
+
+    /// <summary>
+    /// Gets or sets the content to suffix the input component.
+    /// </summary>
+    [Parameter]
+    public virtual RenderFragment? EndTemplate { get; set; }
+
+    /// <summary>
+    /// Gets or sets the size of the checkbox. See <see cref="Components.CheckboxSize"/>
+    /// </summary>
+    [Parameter]
+    public CheckboxSize Size { get; set; } = CheckboxSize.Medium;
+
+    /// <inheritdoc />
+    protected override void OnParametersSet()
+    {
+        base.OnParametersSet();
+        if (HasLabel())
+        {
+            // When the id is not provided, generate a unique id. This allow to use the label for.
+            Id ??= $"checkbox-{Identifier.NewId()}";
+        }
+    }
+
+    /// <summary>
+    /// Parses a string to create the <see cref="Microsoft.AspNetCore.Components.Forms.InputBase{TValue}.Value"/>.
+    /// </summary>
+    /// <param name="value">The string value to be parsed.</param>
+    /// <param name="result">The result to inject into the Value.</param>
+    /// <param name="validationErrorMessage">If the value could not be parsed, provides a validation error message.</param>
+    /// <returns>True if the value could be parsed; otherwise false.</returns>
+    protected override bool TryParseValueFromString(string? value, [MaybeNullWhen(false)] out bool? result, [NotNullWhen(false)] out string? validationErrorMessage)
+    {
+        if (bool.TryParse(value, out var parsedValue))
+        {
+            result = parsedValue;
+            validationErrorMessage = null;
+            return true;
+        }
+
+        result = null;
+        validationErrorMessage = "The provided value is not a valid boolean.";
```

Comment:
When is it possible? 
- Either the value has to be defined by us and if an error is detected we can throw an exception.
- Either this case is possible and we need to use the localisation engine (we cannot hardcode strings)

Reply:
Not resolved